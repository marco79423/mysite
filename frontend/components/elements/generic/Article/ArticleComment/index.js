import ArticleComment from './ArticleComment'

export default ArticleComment
