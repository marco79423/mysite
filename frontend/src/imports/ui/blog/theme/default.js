const colorTable = {
  'cyan-50': '#E0F7FA',
  'cyan-600': '#00ACC1',
  'cyan-700': '#0097A7',
  'cyan-800': '#00838F',
  'cyan-900': '#006064',
  'cyan-A700': '#00B8D4',

  'teal-50':'#E0F2F1',
  'teal-100':'#B2DFDB',
  'teal-200':'#80CBC4',
  'teal-300':'#4DB6AC',
  'teal-400':'#26A69A',
  'teal-500':'#009688',
  'teal-600':'#00897B',
  'teal-700':'#00796B',
  'teal-800':'#00695C',
  'teal-900':'#004D40',
  'teal-A700':'#00BFA5',

  'grey-50': '#FAFAFA',
  'grey-200': '#EEEEEE',
  'grey-300': '#E0E0E0',
  'grey-600': '#757575',
  'grey-700': '#616161',
  'grey-900': '#212121'
}

export default {
  global: {
    fontColor: colorTable['grey-900'],
    link: {
      color: colorTable['cyan-700'],
      hoverColor: colorTable['cyan-900']
    }
  },
  page: {
    background: colorTable['grey-700'],
    header: {
      background: colorTable['teal-400'],
      titleColor: colorTable['grey-50']
    },
    nav: {
      background: colorTable['teal-600'],
      menu: {
        color: '#ECECEC',
        background: colorTable['teal-600'],
        hoverBackground: colorTable['cyan-900'],
        dropdown: {
          background: colorTable['teal-600'],
          borderColor: colorTable['teal-800']
        }
      }
    },
    main: {
      background: colorTable['grey-200'],
      content: {
        article: {
          background: 'white',
          metadata: {
            tip: {
              color: '#FFFFFF',
              background: colorTable['teal-600']
            }
          },
          header: {
            color: colorTable['grey-700']
          },
          block: {
            color: 'white',
            background: colorTable['teal-600'],
            borderColor: '#ccc'
          },
          doctestBlock: {
            color: 'white',
            background: '#6D6D6D',
          },
          note: {
            background: '#ECECEC',
            titleColor: colorTable['teal-800'],
            borderColor: '#ccc'
          },
          table: {
            header: {
              background: colorTable['teal-600'],
            }
          }
        },
        pagination: {
          color: colorTable['grey-200'],
          background: colorTable['teal-700'],
          hoverBackground: colorTable['teal-900'],
          activeBackground: colorTable['teal-900'],
        }
      },
      aside: {
        background: colorTable['grey-200'],
        darkerBackground: colorTable['grey-600'],

        section: {
          titleColor: colorTable['grey-700'],
          borderColor: colorTable['grey-200']
        }
      }
    },
    footer: {
      background: colorTable['teal-600'],
      copyrightColor: '#ECECEC',
    }
  }
}