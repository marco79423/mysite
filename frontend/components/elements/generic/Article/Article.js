import React from 'react'
import styled from 'styled-components'
import {LikeCoinButton} from '@paji-sdk/browser'

import TitleLink from '../TitleLink'
import RstContent from '../RstContent'
import SocialShare from './SocialShare'
import Metadata from './Metadata'
import ArticleComment from './ArticleComment'


const Base = styled.article`
  background: ${props => props.theme.page.main.content.article.background};
  padding: 32px;
`

const Header = styled.header`
  h1 {
    margin: 3px 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`

class Article extends React.PureComponent {

  renderSummary = () => {
    return <RstContent content={this.props.article.summary}/>
  }

  renderContent = () => {
    const {article, commentConfig} = this.props
    return [
      <RstContent key="article-content" content={article.content}/>,
      <LikeCoinButton
        key="likeCoin"
        creatorLikeID={'marco79423'}
        url={article.url}
        style={{marginTop: 48, height: 212, width: '100%'}}/>,
      <SocialShare key="social-share" config={{
        title: article.title,
        shareUrl: article.url,
      }}/>,
      <ArticleComment key="comment" config={commentConfig}/>
    ]
  }

  render() {
    const {article, summaryMode} = this.props
    return (
      <Base>
        <Header>
          <h1><TitleLink href={article.path}>{article.title}</TitleLink></h1>
          <Metadata
            categories={article.categories}
            chickenCount={article.chickenCount}
            date={article.date}
            modifiedDate={article.modifiedDate}/>
        </Header>

        {summaryMode ? this.renderSummary() : this.renderContent()}
      </Base>
    )
  }
}


export default React.memo(Article)
