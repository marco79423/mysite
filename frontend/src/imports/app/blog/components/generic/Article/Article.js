import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import styled from 'styled-components'

import TitleLink from '../TitleLink'
import Loading from '../Loading'
import RstContent from '../RstContent'
import SocialShare from './SocialShare'
import Metadata from './Metadata'
import ArticleComment from './ArticleComment'


const Base = styled.article`
  background: ${props => props.theme.page.main.content.article.background};
  padding: 32px;
`

const Header = styled.header`
  h1 {
    margin: 3px 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`

export default class Article extends React.PureComponent {
  static PropTypes = {
    summaryMode: PropTypes.bool,
    article: ImmutablePropTypes.contains({
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      categories: ImmutablePropTypes.listOf(ImmutablePropTypes.contains({
        slug: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      })).isRequired,
      chickenCount: PropTypes.number.isRequired,
      date: PropTypes.instanceOf(Date).isRequired,
      modifiedDate: PropTypes.instanceOf(Date),
      content: PropTypes.any.isRequired,
      summary: PropTypes.string.isRequired
    }),
    socialConfig: ImmutablePropTypes.map,
    commentConfig: ImmutablePropTypes.map
  }

  renderHeader = () => {
    const {article} = this.props
    return (
      <Header>
        <h1><TitleLink to={`/articles/${article.get('slug')}/`}>{article.get('title')}</TitleLink></h1>
        {this.renderMetadata()}
      </Header>
    )
  }

  renderMetadata = () => {
    const {article} = this.props
    return (
      <Metadata
        categories={article.get('categories')}
        chickenCount={article.get('chickenCount')}
        date={article.get('date')}
        modifiedDate={article.get('modifiedDate')} />
    )
  }

  renderSummary = () => {
    return <RstContent content={this.props.article.get('summary')}/>
  }

  renderContent = () => {
    const {article, socialConfig, commentConfig} = this.props
    return [
      <RstContent key="article-content" content={article.get('content')}/>,
      <SocialShare  key="social-share" config={socialConfig}/>,
      <ArticleComment key="comment" config={commentConfig}/>
    ]
  }

  render () {
    const {article, summaryMode} = this.props
    if (!article) {
      return (
        <Base>
          <Loading/>
        </Base>
      )
    }

    return (
      <Base>
        {this.renderHeader()}
        {summaryMode ? this.renderSummary() : this.renderContent()}
      </Base>
    )
  }
}

