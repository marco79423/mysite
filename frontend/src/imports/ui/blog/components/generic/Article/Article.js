import React from 'react'
import styled from 'styled-components'
import {LikeCoinButton} from '@paji-sdk/browser'

import TitleLink from '../TitleLink'
import Loading from '../Loading'
import RstContent from '../RstContent'
import SocialShare from './SocialShare'
import Metadata from './Metadata'
import ArticleComment from './ArticleComment'


const Base = styled.article`
  background: ${props => props.theme.page.main.content.article.background};
  padding: 32px;
`

const Header = styled.header`
  h1 {
    margin: 3px 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`

export default class Article extends React.PureComponent {

  renderHeader = () => {
    const {article} = this.props
    return (
      <Header>
        <h1><TitleLink to={`/articles/${article.slug}/`}>{article.title}</TitleLink></h1>
        {this.renderMetadata()}
      </Header>
    )
  }

  renderMetadata = () => {
    const {article} = this.props
    return (
      <Metadata
        categories={article.categories}
        chickenCount={article.chickenCount}
        date={article.date}
        modifiedDate={article.modifiedDate}/>
    )
  }

  renderSummary = () => {
    return <RstContent content={this.props.article.summary}/>
  }

  renderContent = () => {
    const {article, socialConfig, commentConfig} = this.props
    return [
      <RstContent key="article-content" content={article.content}/>,
      <LikeCoinButton
        key="likeCoin"
        creatorLikeID={'marco79423'}
        url={socialConfig?.shareUrl}
        style={{marginTop: 48, height: 212, width: '100%'}}/>,
      <SocialShare key="social-share" config={socialConfig}/>,
      <ArticleComment key="comment" config={commentConfig}/>
    ]
  }

  render() {
    const {article, summaryMode} = this.props
    if (!article) {
      return (
        <Base>
          <Loading/>
        </Base>
      )
    }

    return (
      <Base>
        {this.renderHeader()}
        {summaryMode ? this.renderSummary() : this.renderContent()}
      </Base>
    )
  }
}

