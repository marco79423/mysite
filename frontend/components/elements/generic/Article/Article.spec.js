import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {ThemeProvider} from 'styled-components'

import Article from './Article'
import theme from '../../../theme/default'

test('It should render Article component correctly', () => {
  const article = {
    slug: 'slug',
    path: '/articles/slug',
    url: 'host://host/articles/slug',
    title: 'title',
    categories: [{
      slug: 'slug',
      name: 'name'
    }],
    date: new Date(2017, 6, 9),
    chickenCount: 1,
    content: 'content',
    rawSummary: 'raw_summary'
  }
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Article article={article}/>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
