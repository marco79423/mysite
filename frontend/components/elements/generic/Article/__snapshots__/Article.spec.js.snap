// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Article component correctly 1`] = `
.c7 {
  color: #0097A7;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:hover {
  color: #006064;
}

.c2 {
  color: #0097A7;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 2rem;
}

.c2:hover {
  color: #006064;
}

.c10 {
  margin-top: 16px;
  font-size: 1.2rem;
  line-height: 1.6rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
}

.c11 > div:not(:first-child) {
  margin-left: .5rem;
}

.c3 {
  font-size: 0.9rem;
  line-height: 1.6rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c4 {
  display: inline;
}

.c4:not(:first-child) {
  margin-left: 8px;
}

.c5 {
  display: inline;
}

.c6 {
  margin: 0;
  display: inline;
}

.c6:not(:first-child):before {
  content: "/";
  margin: 0 3px;
}

.c8 {
  margin-bottom: -5px;
}

.c8:hover {
  -webkit-transform: rotate(-10deg);
  -ms-transform: rotate(-10deg);
  transform: rotate(-10deg);
}

.c9 {
  display: none;
  position: absolute;
  line-height: 1.2em;
  padding: 4px;
  border-radius: 3px;
  background: #00897B;
  color: #FFFFFF;
  -webkit-transition: opacity .3s ease-out;
  transition: opacity .3s ease-out;
}

.c9:before {
  content: "";
  position: absolute;
  width: 0;
  height: 0;
  left: -15px;
  top: 4px;
  border: 8px solid transparent;
  border-right-color: #00897B;
}

.c0 {
  background: white;
  padding: 32px;
}

.c1 h1 {
  margin: 3px 0;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

@media (max-width:800px) {
  .c9 {
    display: none;
  }
}

<article
  className="c0"
>
  <header
    className="c1"
  >
    <h1>
      <a
        className="c2"
        href="/articles/slug"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        title
      </a>
    </h1>
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        分類：
        <ul
          className="c5"
        >
          <li
            className="c6"
          >
            <a
              className="c7"
              href="/articles/category/slug"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              name
            </a>
          </li>
        </ul>
      </li>
      <li
        className="c4"
      >
        字數：
        <img
          alt="雞"
          className="c8"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          src="/img/chicken.png"
        />
         x 
        1
        <span
          className="c9"
        >
          雞數：計算文長的常見計量單位，一般而言數字大小與文章長度呈正相關
        </span>
      </li>
    </ul>
  </header>
  <div>
    <div
      className="c10 rst-content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "content",
        }
      }
    />
  </div>
  <iframe
    src="//button.like.co/in/embed/marco79423/button?referrer=host://host/articles/slug"
    style={
      Object {
        "border": 0,
        "height": 212,
        "margin": 0,
        "marginTop": 48,
        "width": "100%",
      }
    }
    title="iframe"
  />
  <div
    className="c11"
  >
    <button
      aria-label="facebook"
      className="react-share__ShareButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "inherit",
          "cursor": "pointer",
          "font": "inherit",
          "padding": 0,
        }
      }
      title="title"
    >
      <svg
        height={32}
        viewBox="0 0 64 64"
        width={32}
      >
        <circle
          cx="32"
          cy="32"
          fill="#3b5998"
          r="31"
          style={Object {}}
        />
        <path
          d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
          fill="white"
        />
      </svg>
    </button>
    <button
      aria-label="twitter"
      className="react-share__ShareButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "inherit",
          "cursor": "pointer",
          "font": "inherit",
          "padding": 0,
        }
      }
    >
      <svg
        height={32}
        viewBox="0 0 64 64"
        width={32}
      >
        <circle
          cx="32"
          cy="32"
          fill="#00aced"
          r="31"
          style={Object {}}
        />
        <path
          d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
          fill="white"
        />
      </svg>
    </button>
    <button
      aria-label="linkedin"
      className="react-share__ShareButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "inherit",
          "cursor": "pointer",
          "font": "inherit",
          "padding": 0,
        }
      }
    >
      <svg
        height={32}
        viewBox="0 0 64 64"
        width={32}
      >
        <circle
          cx="32"
          cy="32"
          fill="#007fb1"
          r="31"
          style={Object {}}
        />
        <path
          d="M20.4,44h5.4V26.6h-5.4V44z M23.1,18c-1.7,0-3.1,1.4-3.1,3.1c0,1.7,1.4,3.1,3.1,3.1 c1.7,0,3.1-1.4,3.1-3.1C26.2,19.4,24.8,18,23.1,18z M39.5,26.2c-2.6,0-4.4,1.4-5.1,2.8h-0.1v-2.4h-5.2V44h5.4v-8.6 c0-2.3,0.4-4.5,3.2-4.5c2.8,0,2.8,2.6,2.8,4.6V44H46v-9.5C46,29.8,45,26.2,39.5,26.2z"
          fill="white"
        />
      </svg>
    </button>
  </div>
  <div />
</article>
`;
